core/sdk-core/src/main/java/software/amazon/awssdk/core/client/config/ClientOverrideConfiguration.java

```diff
@@ -230,19 +234,30 @@ public ExecutionAttributes executionAttributes() {
         return executionAttributes;
     }
 
+    /**
+     * The request compression configuration object, which includes options to enable/disable request compression and set the
+     * minimum compression threshold.
+     *
+     * @see Builder#requestCompressionConfiguration(RequestCompressionConfiguration)
+     */
+    public RequestCompressionConfiguration requestCompressionConfiguration() {
+        return requestCompressionConfiguration;
+    }
+
     @Override
     public String toString() {
         return ToString.builder("ClientOverrideConfiguration")
-                .add("headers", headers)
-                .add("retryPolicy", retryPolicy)
-                .add("apiCallTimeout", apiCallTimeout)
-                .add("apiCallAttemptTimeout", apiCallAttemptTimeout)
-                .add("executionInterceptors", executionInterceptors)
-                .add("advancedOptions", advancedOptions)
-                .add("profileFile", defaultProfileFile)
-                .add("profileName", defaultProfileName)
-                .add("scheduledExecutorService", scheduledExecutorService)
-                .build();
+                       .add("headers", headers)
+                       .add("retryPolicy", retryPolicy)
+                       .add("apiCallTimeout", apiCallTimeout)
+                       .add("apiCallAttemptTimeout", apiCallAttemptTimeout)
+                       .add("executionInterceptors", executionInterceptors)
+                       .add("advancedOptions", advancedOptions)
+                       .add("profileFile", defaultProfileFile)
+                       .add("profileName", defaultProfileName)
+                       .add("scheduledExecutorService", scheduledExecutorService)
+                       .add("requestCompressionConfiguration", requestCompressionConfiguration)
+                       .build();
```

Comment:
Did we update hashcode and equals?

Reply:
Hmm, I guess that's an oversight. I guess we can leave it as is for now, not sure if adding them now will break things...